### PR TITLE
rtc changes for C programst

### PR DESCRIPTION
**Pull request by cjatin**
Initial commit to get rtc working with genco

### CODE REVIEW

File: src/hiprtc.cpp [hunk 1]
```diff
@@ -211,7 +211,7 @@ struct _hiprtcProgram {
 
         compile.close();
 
-        if (compile.rdbuf()->exited() &&
+        /*if (compile.rdbuf()->exited() &&
```
Comment (SarbojitAMD) [inline src/hiprtc.cpp]: Commented code, do you want to clean-up later?

File: src/hiprtc.cpp [hunk 2]
```diff
@@ -492,17 +507,20 @@ hiprtcResult hiprtcCompileProgram(hiprtcProgram p, int n, const char** o)
 
     const auto src{p->writeTemporaryFiles(tmp.path())};
 
-    vector<string> args{hipcc, "-shared"};
+    //vector<string> args{hipcc, "-shared"};
```
Comment (SarbojitAMD) [inline src/hiprtc.cpp]: Please remove the commented line

File: src/hiprtc.cpp [hunk 3]
```diff
@@ -492,17 +507,20 @@ hiprtcResult hiprtcCompileProgram(hiprtcProgram p, int n, const char** o)
...
+    vector<string> args{hipcc, "--genco"};
     if (n) args.insert(args.cend(), o, o + n);
 
-    handleTarget(args);
+    // TODO Add targets
```
Comment (SarbojitAMD) [inline src/hiprtc.cpp]: commented code

File: tests/src/hiprtc/hiprtcGetLoweredName.cpp [hunk 4]
```diff
@@ -39,15 +39,15 @@ R"(
 #include <hip/hip_runtime.h>
 
 __device__ int V1; // set from host code
-static __global__ void f1(int *result) { *result = V1 + 10; }
-namespace N1 {
+extern "C" __global__ void f1(int *result) { *result = V1 + 10; }
+/*namespace N1 {
```
Comment (SarbojitAMD) [inline tests/src/hiprtc/hiprtcGetLoweredName.cpp]: dead code may be?

**Comment by AlexVlx** [comment]
I fail to see the point of this, it's fundamentally unclear what you think hiprtc (a runtime API) has to do with genco (a compile-time tool). This should not be merged. If you want to break interfaces that users rely on, do it in a private branch until the point in which you have replacement functionality. @mangupta for awareness

**Pull request closed without merging**